src/datatypes/src/vectors/constant.rs

```diff
@@ -83,6 +84,34 @@ impl ConstantVector {
             self.length,
         )))
     }
+
+    pub(crate) fn take_vector(&self, indices: &UInt32Vector) -> Result<VectorRef> {
+        if indices.is_empty() {
+            return Ok(self.slice(0, 0));
+        }
+        let len = self.len() as u32;
+        indices
+            .iter_data()
+            .map(|idx| {
+                if let Some(idx) = idx {
+                    ensure!(
+                        idx < len,
+                        error::BadArrayAccessSnafu {
+                            index: idx as usize,
+                            size: indices.len()
+                        }
+                    );
+                }
+                Ok(())
+            })
+            .collect::<Result<Vec<_>>>()?;
```

Comment:
[nitpick]: maybe you can use [`arrow::compute::lt_scalar`](https://docs.rs/arrow/33.0.0/arrow/compute/kernels/comparison/fn.lt_scalar.html) here, something like:
```rs
        let indices_arr = indices.to_arrow_array();
        let arr = indices_arr.as_any().downcast_ref::<UInt32Array>().unwrap();
        ensure!(
            !arrow::compute::min_boolean(&arrow::compute::lt_scalar(arr, len).unwrap()).unwrap(),
            BadArrayAccessSnafu{...}
        );
```

Reply:
[suggestion]: to maintain behavior consistency, you can just panic here when accessing out-of-bound elements like [here](https://github.com/GreptimeTeam/greptimedb/pull/1115/files#diff-bfa87fb596051c52504631f7d95b437a2a4b368c4552759e64e3cb5bdda36a9aR23).

Reply:
Nice catch, thanks